### PR TITLE
ADEN-1678 Scroll depth measurement on mobile

### PR DESCRIPTION
**Pull request by nandy-andy**
We would like to track in Google Analytics how much of our pages are being scrolled and if the users scroll to the bottom of our pages. We're going to try a jQuery plugin - [Scroll Depth](http://scrolldepth.parsnip.io/) with my customization.

It's an experiment which we're going to sample. Sampling is hardcoded in the code.

### CODE REVIEW

File: server/views/layout.hbs [hunk 1]
```diff
@@ -78,6 +78,12 @@
 			<script src="/front/vendor/ember-hammer/ember-hammer.js"></script>
 			<script src="/front/vendor/i18next/i18next.js"></script>
 			<script src="/front/vendor/vignette/dist/vignette.js"></script>
+			<!-- @if WIKIA_ENVIRONMENT='dev' -->
+			<script src="/front/vendor/scroll-depth/jquery.scrolldepth.js"></script>
+			<!-- @endif -->
+			<!-- @if WIKIA_ENVIRONMENT!='dev' -->
+			<script src="/front/vendor/scroll-depth/jquery.scrolldepth.min.js"></script>
+			<!-- @endif -->
```
Comment (rogatty) [inline server/views/layout.hbs]: There is no need for that.
https://github.com/Wikia/mercury/blob/dev/gulp/tasks/scripts-front.js#L34 minifies all the scripts.

File: front/scripts/mercury/modules/Trackers/ScrollDepthTracker.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	nonInteraction?: Boolean;
+	unattachEventOnceCacheIsFull?: Boolean;
+	eventHandler: any;
+}
+
+interface ScrollDepth {
+	resetCache(): void;
+}
+
+// TODO: confirm if it's correct way of handling TypeScript error?
```
Comment (rogatty) [inline front/scripts/mercury/modules/Trackers/ScrollDepthTracker.ts]: https://github.com/borisyankov/DefinitelyTyped/search?q=jquerystatic&type=Code

Yes it is.

File: front/scripts/mercury/modules/Trackers/ScrollDepthTracker.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+			this.scrollDepth = jQuery.scrollDepth({
+				detachEventOnceCacheIsFull: false,
+				percentage: false,
+				pixelDepth: false,
+				userTiming: false, // if we want to track it we'll need _trackTiming implementation in GoogleAnalytics.ts
+				minHeight: 750,
+				threshold: 500,
+				eventHandler: (data: ScrollDepthEventData) => {
+					if ( Math.random() * 100 <= this.scrollDepthSample) {
+						Em.Logger.info('Sending scroll depth tracking');
```
Comment (Warkot) [inline front/scripts/mercury/modules/Trackers/ScrollDepthTracker.ts]: Sure, I'm fine with that, but people are becoming aware that they are tracked, but don't know much more about what it is, so they become [paranoid :notes:](http://youtu.be/NZyVZFJGX5g). It's a natural [fear of the dark :notes:](http://youtu.be/Nba3Tr_GLZU). However, we can assume that if someone knows how to open console, then he or she knows what the tracking is. :wink:

**Comment by nandy-andy** [comment]
Thanks for sharing :smiley: As we talked offline: it's only an experiment and we don't really care where the user comes from. We just want to test how "deep" they scroll and according to it we'd like to improve our ads placement :japanese_ogre:

File: front/scripts/mercury/modules/Trackers/ScrollDepthTracker.ts (hunk 3 above)
Comment (rogatty) [inline front/scripts/mercury/modules/Trackers/ScrollDepthTracker.ts]: We're planning to disable Em.Logger for production builds. There is already a ticket https://wikia-inc.atlassian.net/browse/HG-425

File: front/scripts/mercury/modules/Trackers/GoogleAnalytics.ts [hunk 4]
```diff
@@ -119,7 +119,9 @@ module Mercury.Modules.Trackers {
 		 * @arguments set of parameters for ads-related event
 		 */
 		trackAds (): void {
```
Comment (nandy-andy) [inline front/scripts/mercury/modules/Trackers/GoogleAnalytics.ts]: @hakubo at the end I didn't rename this method. I've just made it push only to ad account - it was used only in two places, so changes weren't big.

**Comment by hakubo** [comment]
@nandy-andy Do we have to update trackAd calls to reflect your changes?

**Comment by nandy-andy** [comment]
@hakubo I don't think so. I did the changes **two** places I found usage of `trackAds`:
* [Ads.ts](https://github.com/Wikia/mercury/pull/550/files#diff-d4ef2a33723a7b1872e5642d66167579L73),
* [GoogleAnalytics.js](https://github.com/Wikia/mercury/pull/550/files#diff-08ab8dc4b0ec5dd51b1f8b670002f8d4L94).

I used PhpStorm tool for searching the `trackAds()` calls. If you know about more of them let me know and I'll fix it.

**Comment by nandy-andy** [comment]
ping @hakubo

**Comment by hakubo** [comment]
Ok, lets test it!

**Comment by drets** [comment]
Tested locally, LGTM! Merging.

**Pull request merged**